src/result-list.js

```diff
@@ -12,22 +12,48 @@ function ResultList(retriever, options) {
   this._retriever = retriever;
   this._onFailure = options.onFailure;
   this._onSuccess = options.onSuccess;
-  this._page = options.page || 1;
+  this._page = options.page ? options.page - 1 : 0;
   this._perPage = options.perPage || 20;
+  this._requestPromises = [];
+
+  this._retriever.query({per_page: this._perPage});
 }
 
 ResultList.prototype.next = function() {
-  this._retriever.query({page: this._page, per_page: this._perPage});
   this._page += 1;
+
+  this._retriever.query({page: this._page});
+
+  var promise = this._requestPromises[this._page] ?
+    this._requestPromises[this._page] : this._get();
```

Comment:
Lately, I've avoided expressing my dislike for ternary operators, and even used them in rare cases. But in the case of a ternary operator spanning multiple lines, I can't help but say this should just be an if statement.